src/jllvm/class/Descriptors.cpp

```diff
@@ -38,3 +38,74 @@ std::string jllvm::FieldType::textual() const
         default: llvm_unreachable("Invalid value");
     }
 }
+
+std::string jllvm::FieldType::pretty() const
+{
+    std::string result;
+    if (m_name)
+    {
+        std::replace_copy_if(
+            m_name, m_name + m_size, std::back_inserter(result), [](char c) { return c == '/'; }, '.');
+    }
+    else
+    {
+        llvm::StringRef baseType;
+        switch (m_baseTypeValue)
```

Comment:
nit: Could this code be written more succinct if we had a `std::array<llvm::StringRef` to map the enum to the spelling? Worst case even a `static` hashmap would probably be fine.